### ISSUE TITLE
Recommended SPMC version and settings for my Nvidia Shield Pro? Buffering/lag

### ISSUE THREAD

**Issue by AnarchyMedia85** [koying/SPMC#1151]
I bought a new Nvidia Shield Pro. Put Kodi and constant buffering and such. Was told to use SPMC. I had some buffering. But what I need to know is do I install the Latest Jarvis 16.7.2 on front page or do I install Krypton version that is in the other downloads link?

Also, I seen the Wiki page on here for recommended running Shield & 1080 but some settings given aren't existent on the Krypton SPMC. So I just need to get this all right so I can run a dominate system instead of buffering and what not.

Thank you so much in advance for the help.

**Comment by heymoe** [comment]
I've been running SPMC (v16) on my Shield TV (16GB non-pro) for a while and just installed v17 this past weekend. I followed the Shield & 1080 wiki and couldn't find 1 or 2 options mentioned in the wiki in any of the settings screen such as "vertical blank sync". I've seen that option in the past in Kodi and I'm pretty sure SPMC v16 but can't find it in v17 right now.

Any ways, If you set all other options based on the wiki, you should be good to go for SPMC v16 or v17. 

SPMC v16 from the Play Store has been really solid. I installed v17 (you can have both v16 and v17 installed at the same time) just to test out some of the new features but it is still in testing / alpha mode so likely not as stable as v16 but so far I haven't run into any issues.

In contrast Kodi v17 (Play Store) and Kodi v18 (beta) on the Shield TV seems to be less refined / laggy on the Shield TV when I tried to skip around in a video. 

I would recommend that you hard-wire (Ethernet) your Shield TV instead of using WiFi for its networking. WiFi can be a crap shoot and and would likely cause more issues related to buffering / lag than anything else.

**Comment by AnarchyMedia85** [comment]
Yeah that's what I was thinking of doing is getting the latest Jarvis version because Krypton is still on alpha. But I didn't know if the latest Jarvis has all the right dependencies to run everything correctly.

I'm definitely buying a new router to replace my isp one from my cable provider.

**Comment by heymoe** [comment]
You'll be fine with Jarvis (SPMC v16). I've been using as my primary player on my Shield TV pretty much since I got it and it plays all the standard video formats I use (typically x264/x265) as well as all the different audio formats I use up to Atmos & DTS:X without any problems.

**Comment by AnarchyMedia85** [comment]
So I downloaded Jarvis and omg how smooth and so far better playback. I have a question about Audio. Some movies played with audio and I found only a couple that didn't unless I chose ENABLE PASSTHROUGH. But like I said, it was only a couple movies. Do I permanently turn this option on and leave it or not use it?

**Comment by heymoe** [comment]
If you have a receiver / AVR that can decode audio formats like AC3, DTS, DTS-HD, TrueHD, Atmos, DTS:X, etc.. etc.. then it's best to configure SPMC to pass-through those formats to your receiver to get the best sound out of the Shield TV + SPMC. Only enable the formats your receiver supports / can decode and yes you leave the pass-through settings on all the time.

**Comment by AnarchyMedia85** [comment]
No receiver or surround sound. Just the shield and my TV.so should I just leave those settings alone and keep pass-through on or off

**Comment by heymoe** [comment]
Your TV might support AC3 and/or core DTS decoding. What format did you have to enable pass-through on to get sound?

**Comment by AnarchyMedia85** [comment]
Hmm I'm not too sure. I hit the pass through option during video play. I don't think it told me. Where should I look for that?

**Comment by heymoe** [comment]
I use this program to tell me the details about media files such as video and audio formats.

[MediaInfo](https://mediaarea.net/en/MediaInfo)

Load up the file you had to enable pass-through on and see what it says.

**Comment by AnarchyMedia85** [comment]
I have spmc on shield so idk how to pull up that site and download it and test it all at once. Would have to be on my laptop but then I don't see it helping to know what is happening on the stream on Kodi lol.

I need to somehow find a way to see what my TV can handle. Turn off PASSTHROUGH and receiver stuff until then.

**Comment by heymoe** [comment]
Ahh,, you're streaming media and not playing a file you have on your Shield TV / network share or what not. 

You can try pulling up the Codeinfo screen inside of Kodi / SPMC while you're playing a video / stream to see what the audio and video format the video is using.

http://kodi.wiki/view/Codecinfo

**Comment by heymoe** [comment]
The Info you get from that MediaInfo program or the Codecinfo screen will tell you what the audio format is. So on the videos you're playing where you don't get audio unless you turn on Passthrough, you can find out what audio format that is (for example, AC3) and then you can go into SPMC settings and enable passthough for AC3 so any video you watch that has AC3 will automatically be passed through instead of you having to toggle it manually.

**Comment by heymoe** [comment]
This all would be related to the this wiki doc:

https://github.com/koying/SPMC/wiki/Recommended-passthrough-audio-settings-for-SPMC%2C-NVIDIA-Shield-TV-and-AV-Receivers

**Comment by AnarchyMedia85** [comment]
Okay that makes perfect sense. So I will turn off the settings and test it with that and see. I appreciate the help. Keep ya updated.

**Comment by AnarchyMedia85** [comment]
![img_20180106_021748983](https://user-images.githubusercontent.com/35089746/34637614-e783ccbe-f287-11e7-9a22-c9249e18ec54.jpg)

Okay here is a shot of what it says when I click "o" to view my audio when watching this movie. By this, what should I choose in the audio settings in SPMC?

**Comment by AnarchyMedia85** [comment]
![img_20180106_021748983](https://user-images.githubusercontent.com/35089746/34647220-8f5a358c-f34a-11e7-87b7-f12f7c532a2c.jpg)

Okay here is a shot of what it says when I click "o" to view my audio when watching this movie. By this, what should I choose in the audio settings in SPMC?

**Comment by heymoe** [comment]
Looks like that video / stream is using AAC for its audio codec / format. It would not be a typical passthrough format but in theory if the audio track is sent raw to the downstream device (TV / receiver) and that device knows how to decode it, it would work.

The common passthrough formats that would show up in the "o" screen which you can configure in Settings -> System -> Audio would be: ac3, dts (dca), truehd. I'm not 100% sure what Dolby Digital Plus shows up as on that screen, maybe ddp, dd+ or ac3+

Is the video from your screenshot not provide you audio unless you toggle passthrough? This would seem odd to me if so.

**Comment by AnarchyMedia85** [comment]
No audio has been working perfectly lately. But I noticed some weird static a little so I needed to know exactly what to turn on and off. I set everything back to default and just turn off passthrough
